src/store/modules/auth.js

```diff
@@ -155,5 +155,11 @@ export default {
         context.commit('clearAuth');
       }
     },
+    disconnectSocket() {
+      io.socket.disconnect();
```

Comment:
Despite feeling like an anti pattern, this does collocate everything and prevent s as more complicated test solve— glad it worked!